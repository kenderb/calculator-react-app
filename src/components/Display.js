import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Display extends Component {
  constructor({ results }) {
    super({ results });
    this.state = { calculatoResults: '0' };
  }

  render() {
    const { calculatoResults } = this.state;
    return (
      <div>
        <p>
          results:
        </p>
        {calculatoResults}
      </div>
    );
  }
}

Display.propTypes = {
  results: PropTypes.string.isRequired,
};
export default Display;
