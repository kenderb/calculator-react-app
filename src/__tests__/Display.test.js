import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../components/Display';

test('renders correctly', () => {
  const tree = create(<Display results="results" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays 0', () => {
  render(<Display results="0" />);
  const number = screen.getByText('0');
  expect(number).toBeInTheDocument();
});
