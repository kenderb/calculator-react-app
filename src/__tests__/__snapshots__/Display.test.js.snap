// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ hXDNAS"
>
  results
</div>
`;
